ts/src/profiler.ts

```diff
@@ -268,25 +330,31 @@ export class Profiler extends common.ServiceObject {
       body: reqBody,
       json: true,
     };
-
+    
     this.logger.debug(`Attempting to create profile.`);
-    const [prof, response] = await this.request(options);
-    if (!hasHttpStatusCode(response)) {
-      throw new Error('Server response missing status information.');
-    }
-    if (isErrorResponseStatusCode(response.statusCode)) {
-      let message: number|string = response.statusCode;
-      // tslint:disable-next-line: no-any
-      if ((response as any).statusMessage) {
-        message = response.statusMessage;
-      }
-      throw new Error(message.toString());
-    }
-    if (!isRequestProfile(prof)) {
-      throw new Error(`Profile not valid: ${JSON.stringify(prof)}.`);
-    }
-    this.logger.debug(`Successfully created profile ${prof.profileType}.`);
-    return prof;
+    return new Promise<RequestProfile>((resolve, reject) => {
+      this.request(
+          options,
+          (err: Error, prof: object, response: http.ServerResponse) => {
```

Comment:
It might make sense to make this code fragment a separate function like `responseToProfileOrError`.

Reply:
Done.

Reply:
Done.